source/base/partitioner.cc

```diff
@@ -277,7 +278,8 @@ namespace Utilities
               import_targets_temp.push_back(std::pair<unsigned int,
                                             unsigned int> (i, receive_buffer[i]));
             }
-        import_targets_data = import_targets_temp;
+
+        swap(import_targets_temp, import_targets_data);
```

Comment:
I think the cleaner approach would actually be to use
```
  import_targets_data = std::move(import_targets_temp);
```
or something similar. This makes it clear to the reader which variable survives.